src/components/TodoItem.tsx

```diff
@@ -0,0 +1,115 @@
+import { FC, FormEvent, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  todo: Todo;
+  onDelete?: () => void;
+  onToggle?: () => void;
+  onRename?: (todo: Todo) => void;
+  isLoading: boolean;
+}
+
+export const TodoItem: FC<Props> = ({
+  todo,
+  onDelete = () => {},
+  onToggle = () => {},
+  onRename = () => {},
+  isLoading,
+}) => {
+  const [isRemaining, setIsRemaining] = useState(false);
```

Comment:
```suggestion
  const [isRenaming, setIsRenaming] = useState(false);
```
I guess you mean this